core/server/api_container/server/startosis_engine/kurtosis_starlark_framework/kurtosis_plan_instruction/kurtosis_plan_instruction.go

```diff
@@ -59,7 +69,14 @@ func (builtin *KurtosisPlanInstructionWrapper) CreateBuiltin() func(thread *star
 		if builtin.instructionPlanMask.HasNext() {
 			_, scheduledInstructionPulledFromMaskMaybe = builtin.instructionPlanMask.Next()
 			if scheduledInstructionPulledFromMaskMaybe != nil {
-				instructionResolutionStatus = instructionWrapper.TryResolveWith(scheduledInstructionPulledFromMaskMaybe.GetInstruction(), builtin.enclaveComponents)
+
+				scheduledInstructionUuid := scheduledInstructionPulledFromMaskMaybe.GetUuid()
```

Comment:
Interesting, what guarantees here that the UUID will be the same? Were you able to test that idempotent runs still work? Reading this, I got worried that we might loose the UUID of the scheduled instructions, but maybe I'm wrong. Quickly checking runs are still idempotent with an APIC restart in the middle would clear that doubt

Reply:
After thoughts, I think it's fine actually. But still, curious to know whether you got it to work